signer/src/storage/memory/write.rs

```diff
@@ -172,21 +173,14 @@ impl DbWrite for SharedStore {
         Ok(())
     }
 
-    async fn write_stacks_block_headers(
-        &self,
-        blocks: Vec<model::StacksBlock>,
-    ) -> Result<(), Error> {
+    async fn write_stacks_block_headers(&self, headers: &TenureBlockHeaders) -> Result<(), Error> {
+        let block_iter = headers.clone().into_iter();
         let mut store = self.lock().await;
         store.version += 1;
 
-        blocks.iter().for_each(|block| {
-            store.stacks_blocks.insert(block.block_hash, block.clone());
-            store
-                .bitcoin_anchor_to_stacks_blocks
```

Comment:
We use `bitcoin_anchor_to_stacks_blocks` in `get_stacks_chain_tip`, I would expect some test to fail since we are not populating it in here anymore 🤔